File: LeetCode/python3/611.py
class Solution:
    def triangleNumber(self, nums):
        """
        :type nums: List[int]
        :rtype: int
        """
        nums.sort()
        size, total = len(nums), 0
        for i in range(2, size):
            if nums[i] == 0:
                continue
            j, k = 0, i - 1
            while j < k:
                if nums[i] + nums[j] > nums[k] and nums[i] - nums[j] < nums[k]:
                    total += k - j
                    k -= 1
                else:
                    j += 1
        return total

File: LeetCode/cpp/611.cpp
class Solution {
public:
    int triangleNumber(vector<int> &nums) {
        int num = 0, size = nums.size();
        sort(nums.begin(), nums.end());
        for (int i = 2; i < size; ++i) {
            if (nums[i] == 0)
                continue;
            int j = 0, k = i - 1;
            while (j < k) {
                if (nums[j] + nums[k] > nums[i] && nums[j] - nums[k] < nums[i]) {
                    num += k - j;
                    --k;
                } else
                    ++j;
            }
        }
        return num;
    }
};
